src/internal/_eqDeep.js

```diff
@@ -0,0 +1,45 @@
+var eq = require('../eq');
+var keys = require('../keys');
+var type = require('../type');
+
+// The algorithm used to handle cyclic structures
+// is heavily inspired by underscore.js'
+// implementation
+module.exports = function _eqDeep(a, b, stackA, stackB) {
+    for (var idx = stackA.length - 1; idx >= 0; idx--) {
```

Comment:
House style only uses `while` loops.  While this has never been a formal requirement of Ramda code, I think we're pretty loathe to change it.

Usually:

    var idx = -1;
    while (++idx < bound) { /* ... */ }